### DISCUSSION TITLE
Execute migration in transaction

### DISCUSSION THREAD

**Discussion by gleb-chipiga** [SeaQL/sea-orm#1324] [Ideas]
## Motivation

I think that in case of an error during the migration, it will be useful to automatically restore the database to a previous consistent state. Thus, a failed migration will not affect the operation of the application.

## Proposed Solutions

To do this, we need to run the migration and update the seaql_migrations table in a transaction.

**Comment by billy1624** [comment]
Hey @gleb-chipiga, I think it's hard to achieve "atomic migration" because not all database engine support it. Please check previous discussion at https://github.com/SeaQL/sea-orm/discussions/1059

**Reply by gleb-chipiga** [reply]
In sqlx, atomic migrations are supported at least for PostgreSQL. Although the set of database engines there is more than in SeaORM. I think the sqlx approach can be applied to SeaORM as well.

**Comment by billy1624** [comment]
Hey @gleb-chipiga, tracking issue https://github.com/SeaQL/sea-orm/issues/1378
